PortalFetch/crawler_data_process.py

```diff
@@ -4,65 +4,157 @@
         for tr in trs:
             ui = []
             for td in tr:
-                ui.append(td.string)
-            ulist.append(ui)
+                if td == '\n':
+                    continue
+                elif td.string:
+                    ui.append(str(td.string))
+                else:
+                    ui.append(td)
+            if len(ui) == 19:
+                rustCourseList.append(ui[:18])
+        return rustCourseList
 
-    def __save_contents(self, filename, firstline, urlist):
+    def __getList(self, courseList, html):
+        """
+        Get course list from html.
+
+        Input: html is html of the courses, courseList is an empty list []
+        Output : courseList will include all the courses list in text
+        Parameters: List, String
+        Returns: None
         """
-        Save course result to a .csv file.
+        ui = self.__getRustContents(html)
+        for course in ui:
+            temp = []
+            for ele in course:
+                if type(ele) is not str:
+                    soup = BeautifulSoup(str(ele), 'lxml')
+                    ele = soup.get_text()
+                temp.append((ele))
+            courseList.append(temp)
 
-        input : urlist is the courses list we get from get_contents
-        output : save courses in a file "DeAnza2020spring.csv"
+    def __deputyList(self, courseList):
+        """
+        Deputy courseList to a json file.
 
-        Parameters: String, String, List
+        Input: courseList is courses list from __getContents
+        Output : .json file  will include all the courses information
+        Parameters: List
         Returns: None
         """
-        with open(filename, 'w', encoding='utf-8', newline='') as f:
-            writer = csv.writer(f)
-            writer.writerow([firstline])
-
-            for i in range(8, len(urlist)-1):
-                if len(urlist[i]) >= 35:
-                    writer.writerow([urlist[i][1], urlist[i][3], urlist[i][5],urlist[i][7], urlist[i][9], urlist[i][11],
-                                urlist[i][13], urlist[i][15], urlist[i][17],urlist[i][19], urlist[i][21], urlist[i][23],
-                                urlist[i][25], urlist[i][27], urlist[i][29],urlist[i][31], urlist[i][33], urlist[i][35]]
-                                )
+        dic = {}
+        title = ['Select', 'CRN', 'Coreq', 'Subj', 'Crse', 'Sec', 'Cmp', 'Cred', 'Title', 'Days', 'Time', 'Act', 'Rem',
+                'WL Rem', 'Instructor', 'Date (MM/DD)', 'Location', 'Attribute', 'lab']
+        for i in range(len(courseList)):
+            # Create a new subject
+            if courseList[i][0] == 'Select':
+                subj = courseList[i+1][3]
+                d = []
+                dic[subj] = d
+            else:
+                # deputy one line of course information
+                if courseList[i][0] != '\xa0':
+                    di = {}
+                    self.__deputyCourseLine(title, courseList[i], di)
```

Comment:
The filename should be changed to end with ```.json```, otherwise LGTM. The next step of this repo should be focusing on unit testing and add some more forms of data collection.